pkg/bwagreement/server.go

```diff
@@ -50,15 +50,15 @@ type DB interface {
 
 // Server is an implementation of the pb.BandwidthServer interface
 type Server struct {
-	db     DB
-	NodeID storj.NodeID
-	logger *zap.Logger
+	db       DB
+	identity *identity.FullIdentity
```

Comment:
all this file needs is just storj.NodeID and its public key. Passing FullIdentity, I am thinking we are passing way more information that needed. Shouldnt we just keep it to only needed data ?